src/packages/utils/tests/json-rpc-data.test.ts

```diff
@@ -0,0 +1,187 @@
+import assert from "assert";
+import {Data} from "..";
+
+describe("json-rpc-data", () => {
+  const validValues = [ "0x", "0x1", "0x1234", Buffer.from([]), Buffer.from([0x12,0x34]) ];
+  const invalidValues: any[] = [ "1234", 1234n, NaN/*, undefined, null, 1234, [], {}, "0x-1234"*/ ]; // todo: this should be addressed in rewrite of json-rpc-data
+                                                                                                     // See related https://github.com/trufflesuite/ganache/issues/2728
```

Comment:
```suggestion
                                                                                                     // See related https://github.com/trufflesuite/ganache/labels/json-rpc%20refactor
```
Since it's more a collection of issues rather than a specific issue, I made a label in github and added that label to the relevant issues.